llvm/include/llvm/Cheerp/GlobalDepsAnalyzer.h

```diff
@@ -52,7 +52,7 @@ class GlobalDepsAnalyzer
 	typedef llvm::SmallVector<const llvm::Use *, 8> SubExprVec;
 	typedef std::unordered_multimap< const llvm::GlobalVariable *, SubExprVec > FixupMap;
 
-	GlobalDepsAnalyzer(MATH_MODE mathMode = NO_BUILTINS, bool llcPass = false);
+	GlobalDepsAnalyzer(MATH_MODE mathMode = NO_BUILTINS, bool llcPass = false, bool preserveFree = false);
```

Comment:
let's just use the global variable PreserveFree directly, instead of passing arguments around